File: spring-web-java/src/main/java/spring/web/java/presentation/dto/response/OrderResponse.java
package spring.web.java.presentation.dto.response;

import java.time.Instant;
import java.util.List;

import spring.web.java.domain.model.entity.Order;
import spring.web.java.domain.model.enums.OrderStatus;

public record OrderResponse(
	Long id,
	String name,
	Instant orderedAt,
	OrderStatus status,
	List<OrderItemResponse> orderItems
) {
	public OrderResponse(Order order) {
		this(
			order.getId(),
			order.getMember().getName(),
			order.getOrderedAt(),
			order.getStatus(),
			order.getOrderItems().stream()
				.map(OrderItemResponse::new)
				.toList()
		);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/presentation/dto/response/OrderResponse.kt
package spring.web.kotlin.presentation.dto.response

import spring.web.kotlin.domain.model.entity.Order
import spring.web.kotlin.domain.model.enums.OrderStatus
import java.time.Instant

data class OrderResponse(
    val id: Long,
    val name: String,
    val orderedAt: Instant,
    val status: OrderStatus,
    val orderItems: List<OrderItemResponse>
) {
    constructor(order: Order) : this(
        id = checkNotNull(order.id),
        name = order.member.name,
        orderedAt = order.orderedAt,
        status = order.status,
        orderItems = order.orderItems.map { OrderItemResponse(it) }
    )
}